main.go

```diff
@@ -4,17 +4,47 @@ import (
 	"fmt"
 	"github.com/shirou/gopsutil/cpu"
 	"github.com/shirou/gopsutil/mem"
+	"html/template"
+	"log"
 	"net/http"
 )
 
+var tpl *template.Template
+
+func init() {
+	tpl = template.Must(template.ParseFiles("index.gohtml"))
+}
+
+type monitor struct {
+	CpuPercent []float64
+	MemPercent *mem.VirtualMemoryStat
+	Message    string
+}
+
+var data monitor
+
 func index(w http.ResponseWriter, r *http.Request) {
-	cpuPercent, _ := cpu.Percent(0, false)
-	memPercent, _ := mem.VirtualMemory()
-	message := ""
-	if cpuPercent[0] > 2 || memPercent.UsedPercent > 10 {
-		message = "High CPU or Memory utilization. Please scale up!!"
+
+	data.CpuPercent, _ = cpu.Percent(0, false)
+	data.MemPercent, _ = mem.VirtualMemory()
```

Comment:
`cpu.Percent` and `mem.VirtualMemory` functions are called sequentially, which might lead to performance bottlenecks, especially in high-traffic scenarios

Will use concurrency features (goroutines), to execute these tasks concurrently and improve responsiveness

Reply:
Including go routiine
```suggestion
	var wg sync.WaitGroup
	wg.Add(2)

	go func() {
		defer wg.Done()
		cpuPercent, err := cpu.Percent(0, false)
		if err != nil {
			log.Println("Error fetching CPU percent:", err)
			return
		}
		data.CpuPercent = cpuPercent
	}()
	data.MemPercent, _ = mem.VirtualMemory()
```